File: pattern-jwt-server/src/main/java/fm/pattern/jwt/server/service/RoleService.java
package fm.pattern.jwt.server.service;

import java.util.List;

import fm.pattern.jwt.server.model.Role;
import fm.pattern.microstructure.Result;

public interface RoleService {

	Result<Role> create(Role role);

	Result<Role> update(Role role);

	Result<Role> delete(Role role);

	Result<Role> findById(String id);

	Result<List<Role>> list();

}

File: tokamak-console/app/src/services/RoleService.js
import RestService from "./RestService.js"

var RoleService = {
        
    list() {
        return RestService.list("/v1/roles");
    },
    
    create(name, description) {
        return RestService.post("/v1/roles", { name:name, description:description });
    },

    update(id, name, description) {
        return RestService.put("/v1/roles/" + id, { name:name, description:description });
    },

    delete(role) {
        return RestService.delete("/v1/roles/" + role.id, role);
    }      
    
}

export default RoleService;